### PR TITLE
Various Fixes and Pendulum Dynamics

### PR DESCRIPTION
**Pull request by include4eto**
- Fixed `python3` bindings
- Dynamics solvers now record iteration 0 with the initial cost
- Added the pendulum example
- Added a collision example with the quadrotor robot

### CODE REVIEW

File: exotations/dynamics_solvers/exotica_pendulum_dynamics_solver/include/exotica_pendulum_dynamics_solver/pendulum_dynamics_solver.h [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    /// @param x The state vector.
+    /// @param u The control input
+    /// @return The derivative of the dynamics function w.r.t. u evaluated at (x, u).
+    Eigen::MatrixXd fu(const StateVector& x, const ControlVector& u) override;
+
+private:
+    double g_ = 9.81;  ///!< Gravity (m/s^2)
+    double m_ = 1.0;   ///!< Mass at the end of the pendulum
+    double l_ = 1.0;   ///!< Length of the arm
+    double b_ = 0.0;   ///!< Friction coefficient
```
Comment (wxmerkt) [inline exotations/dynamics_solvers/exotica_pendulum_dynamics_solver/include/exotica_pendulum_dynamics_solver/pendulum_dynamics_solver.h]: This one probably should be configurable within the initialiser

File: exotations/dynamics_solvers/exotica_pendulum_dynamics_solver/package.xml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  <buildtool_depend>catkin</buildtool_depend>
+  <build_depend>exotica_core</build_depend>
+  <build_depend>roscpp</build_depend>
+  <build_export_depend>exotica_core</build_export_depend>
+  <build_export_depend>roscpp</build_export_depend>
+  <exec_depend>exotica_core</exec_depend>
+  <exec_depend>roscpp</exec_depend>
+  <!-- This is not yet upstream. PR link: https://github.com/ros/rosdistro/pull/21332 -->
+  <!-- <test_depend>python-autograd</test_depend> -->
```
Comment (wxmerkt) [inline exotations/dynamics_solvers/exotica_pendulum_dynamics_solver/package.xml]: This is indeed upstream, but as `autograd-pip`

File: exotations/solvers/exotica_ddp_solver/src/abstract_ddp_solver.cpp [hunk 3]
```diff
@@ -54,9 +63,9 @@ void AbstractDDPSolver::Solve(Eigen::MatrixXd& solution)
     if (debug_) HIGHLIGHT_NAMED("DDPSolver", "Running DDP solver for max " << GetNumberOfMaxIterations() << " iterations");
 
     double last_cost = 0, global_best_cost = 0;
-    int last_best_iteration = 0;
+    int last_best_iteration = 1;
```
Comment (wxmerkt) [inline exotations/solvers/exotica_ddp_solver/src/abstract_ddp_solver.cpp]: What if the 0th guess was better than the first?

File: exotica_examples/resources/robots/pendulum.urdf [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+    <material name="red">
+        <color rgba="0.9 0.7 0.2 1"/>
+    </material>
+
+    <!-- links -->
+    <link name="base_link">
+        <visual>
+            <geometry>
+                <sphere radius="0.05"/>
```
Comment (wxmerkt) [inline exotica_examples/resources/robots/pendulum.urdf]: Could use a cylinder like a hinge joint, for aesthetics

File: exotica_examples/resources/robots/pendulum.urdf [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+            </geometry>
+            <material name="blue"/>
+        </visual>
+    </link>
+
+    <link name="arm">
+        <visual>
+            <geometry>
+                <box size="0.015 0.015 1"/>
```
Comment (wxmerkt) [inline exotica_examples/resources/robots/pendulum.urdf]: Box or cylinder, both fine. Cylinder may be more aesthetic ;-)

File: exotica_examples/resources/robots/pendulum.urdf [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+            </geometry>
+            <origin rpy="0 0 0" xyz="0 0 -0.5"/>
+            <material name="blue"/>
+        </visual>
+    </link>
+
+    <link name="mass">
+        <visual>
+            <geometry>
+                <sphere radius="0.1"/>
```
Comment (wxmerkt) [inline exotica_examples/resources/robots/pendulum.urdf]: Or a box...

File: exotations/dynamics_solvers/exotica_pendulum_dynamics_solver/include/exotica_pendulum_dynamics_solver/pendulum_dynamics_solver.h (hunk 1 above)
Comment (include4eto) [reply exotations/dynamics_solvers/exotica_pendulum_dynamics_solver/include/exotica_pendulum_dynamics_solver/pendulum_dynamics_solver.h]: Done

File: exotations/dynamics_solvers/exotica_pendulum_dynamics_solver/src/pendulum_dynamics_solver.cpp [hunk 7]
```diff
@@ -46,6 +46,8 @@ void PendulumDynamicsSolver::AssignScene(ScenePtr scene_in)
     // TODO: This is a terrible check (not against name etc.), but can stop _some_ mismatches between URDF/model and dynamics
     if (num_positions_in != 1)
         ThrowPretty("Robot model may not be a Pendulum.");
+
+    b_ = parameters_.FrictionCoefficient;
```
Comment (wxmerkt) [inline exotations/dynamics_solvers/exotica_pendulum_dynamics_solver/src/pendulum_dynamics_solver.cpp]: May want to do a sanity check on that... think, negative friction etc.

**Pull request merged**